core/pow/engine_test.go

```diff
@@ -149,13 +151,105 @@ func TestMempoolTidRejection(t *testing.T) {
-	require.Equal(t, uint(60), calculateExpectedDifficulty(20, 5, 3))   // 3 * 20
-	require.Equal(t, uint(100), calculateExpectedDifficulty(20, 5, 5))  // 5 * 20
-	require.Equal(t, uint(121), calculateExpectedDifficulty(20, 5, 6))  // 5 * 20 + 21
-	require.Equal(t, uint(184), calculateExpectedDifficulty(20, 5, 9))  // 5 * 20 + 4 * 21
-	require.Equal(t, uint(205), calculateExpectedDifficulty(20, 5, 10)) // 5 * 20 + 5 * 21
-	require.Equal(t, uint(430), calculateExpectedDifficulty(20, 5, 20)) // 5 * 20 + 5 * 21 + 5 * 22 + 5 * 23
-	require.Equal(t, uint(478), calculateExpectedDifficulty(20, 5, 22)) // 5 * 20 + 5 * 21 + 5 * 22 + 5 * 23 + 2 * 24
+	type args struct {
+		spamPowDifficulty         uint
+		spamPoWNumberOfTxPerBlock uint
+		seenTx                    uint
+	}
+
+	tests := []struct {
+		name           string
+		args           args
+		wantTotal      uint
+		wantDifficulty uint
+	}{
+		{
+			name: "3 transactions",
+			args: args{
+				spamPowDifficulty:         20,
+				spamPoWNumberOfTxPerBlock: 5,
+				seenTx:                    3,
+			},
+			wantTotal:      60, // 3 * 20
+			wantDifficulty: 20,
+		},
+		{
+			name: "5 transactions",
+			args: args{
+				spamPowDifficulty:         20,
+				spamPoWNumberOfTxPerBlock: 5,
+				seenTx:                    5,
+			},
+			wantTotal:      100, // 5 * 20
+			wantDifficulty: 21,
+		},
+		{
+			name: "6 transactions",
+			args: args{
+				spamPowDifficulty:         20,
+				spamPoWNumberOfTxPerBlock: 5,
+				seenTx:                    6,
+			},
+			wantTotal:      121, // 5 * 20 + 21
+			wantDifficulty: 21,
+		},
+		{
+			name: "9 transactions",
+			args: args{
+				spamPowDifficulty:         20,
+				spamPoWNumberOfTxPerBlock: 5,
+				seenTx:                    9,
+			},
+			wantTotal:      184, // 5 * 20 + 4 + 21
+			wantDifficulty: 21,
+		},
+		{
+			name: "10 transactions",
+			args: args{
+				spamPowDifficulty:         20,
+				spamPoWNumberOfTxPerBlock: 5,
+				seenTx:                    10,
+			},
+			wantTotal:      205, // 5 * 20 + 5 * 21
+			wantDifficulty: 22,
+		},
+		{
+			name: "20 transactions",
+			args: args{
+				spamPowDifficulty:         20,
+				spamPoWNumberOfTxPerBlock: 5,
+				seenTx:                    20,
+			},
+			wantTotal:      430, // 5 * 20 + 5 * 21 + 5 * 22 + 5 * 23
+			wantDifficulty: 24,
+		},
+		{
+			name: "22 transactions",
+			args: args{
+				spamPowDifficulty:         20,
+				spamPoWNumberOfTxPerBlock: 5,
+				seenTx:                    22,
+			},
+			wantTotal:      478, // 5 * 20 + 5 * 21 + 5 * 22 + 5 * 23 + 2 * 24
+			wantDifficulty: 24,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			gotTotal, gotDifficulty := calculateExpectedDifficulty(tt.args.spamPowDifficulty, tt.args.spamPoWNumberOfTxPerBlock, tt.args.seenTx)
+			require.Equal(t, tt.wantTotal, gotTotal)
+			require.Equal(t, tt.wantDifficulty, gotDifficulty)
+		})
+	}
+
+	// require.Equal(t, uint(60), calculateExpectedDifficulty(20, 5, 3))   // 3 * 20
```

Comment:
Should we get rid of this commented out code? The for loop above seems to be testing against the data included.

Reply:
we should